### PR TITLE
Tools: allow to run sim_vehicle.py without installing pymavlink

### PR DESCRIPTION
**Pull request by khancyr**

### CODE REVIEW

File: Tools/autotest/pysim/util.py [hunk 1]
```diff
@@ -19,7 +19,13 @@
 
 import pexpect
 
-from pymavlink.rotmat import Vector3, Matrix3
+pysim_dir = os.path.dirname(os.path.realpath(__file__))
+root_dir = os.path.realpath(os.path.join(pysim_dir, '../../..'))
+try:
+    from pymavlink.rotmat import Vector3, Matrix3
+except ImportError:
```
Comment (tridge) [inline Tools/autotest/pysim/util.py:26]: i'm surprised this works. how will mavproxy.py find pymavlink?

Comment (peterbarker) [reply Tools/autotest/pysim/util.py:26]: > i'm surprised this works. how will mavproxy.py find pymavlink?

I image this is for running with the `--no-mavproxy` command-line option

Comment (khancyr) [reply Tools/autotest/pysim/util.py:26]: Yes this intended to run without mavproxy

Comment (khancyr) [reply Tools/autotest/pysim/util.py:26]: For notice there is already the same thing into sim_vehicle.py , som dependencies have changed some we need the exception here too now

**Comment by peterbarker** [comment]
It's nice that this works.

It's not necessarily a blocker, but this will break my workflows.   I edit pymavlink sourcecode in a separate checkout unrelated to ArduPilot, and install it from there, pointing at the ArduPilot checkout's message set.

Could be some oddities involved here.  This means if you have installed MAVProxy it will be using a different pymavlink (and set of valid messages) to `sim_vehicle.py`.  Speaking of which - in this system here how do you update the message bindings available to `sim_vehicle.py`?  Which bindings does it use?

**Comment by khancyr** [comment]
my principal target for this is container, so everything is static. That just give way to use sim_vehicle.py as launcher instead of calling the binary directly. 

What do you call `message bindings available to sim_vehicle.py` ? sim_vehicle.py doesn't do anything with mavlink. 

For the oddities, that is already broken in our installation. 
We install pymavlink and mavproxy with the install script from pip, but never update them. So you could update ardupilot or change branch and have some mavlink msg on ardupilot that don't match mavproxy and same for pymavlink. ArduPilot build system is using msg definition and generator from the modules/ directory. So this PR just allow to do the same with sim_vehicle.py. Ideally, there is better chance it works than a pip installed pymavlink that could be at an update version and have a different rotmat module.

In anycase Mavproxy will need a pymavlink version installed. So if you don't have pymavlink and try to use mavproxy, it will failed
```
SIM_VEHICLE: "mavproxy.py" "--out" "127.0.0.1:14550" "--out" "127.0.0.1:14551" "--master" "tcp:127.0.0.1:5760" "--sitl" "127.0.0.1:5501" "--console"
RiTW: Starting Rover : /home/khancyr/Workspace/ardupilot/build/sitl/bin/ardurover -S --model rover --speedup 1 --slave 0 --defaults Tools/autotest/default_params/rover.parm --sim-address=127.0.0.1 -I0
Traceback (most recent call last):
  File "/usr/local/bin/mavproxy.py", line 39, in <module>
    from MAVProxy.modules.lib import rline
  File "/usr/local/lib/python3.10/dist-packages/MAVProxy/modules/lib/rline.py", line 8, in <module>
    from pymavlink import mavutil
ModuleNotFoundError: No module named 'pymavlink'
SIM_VEHICLE: MAVProxy exited
SIM_VEHICLE: Killing tasks
```
The PR only work with --no-mavproxy

**Comment by tridge** [comment]
this would break my workflow too - I am often working on a pymavlink PR in a separate repo, and I install there and then test with sim_vehicle.py

**Comment by tridge** [comment]
I suspect we could remove the dependence on pymavlink completely in sim_vehicle.py pretty easily

**Comment by peterbarker** [comment]
> rework to remove pymavlink dependency on utils.py . I moved the use of Vector3 and Matrix3 into a new jsb_wind.py. This hold the wind simulation for jsbsim now.

Cool.

Does jsbsim still work?

**Comment by khancyr** [comment]
no idea, I have never used it...

**Comment by khancyr** [comment]
I tried jsbsim on wsl2 following https://ardupilot.org/dev/docs/sitl-with-jsbsim.html and it crash the same way as master with an issue reset.xml. 
So that all I can test for jsbsim. Python code is not crashing so the import is working, so we should be fine with those changes

**Comment by tridge** [comment]
I think we can remove the python jsb_sim code completely - it is unused

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**